pkg/eth/eth.go

```diff
@@ -280,9 +447,68 @@ func (eth *EthService) refreshEndpoints(ctx context.Context) error {
 				Blocknumber:    node.BlockNumber.Int64(),
 			},
 		); err != nil {
-			return fmt.Errorf("could not insert registered endpoint into eth indexer db: %w", err)
+			return fmt.Errorf("could not insert registered endpoint into eth indexer db: %v", err)
+		}
+
+		if _, ok := allServiceProviders[node.Owner.Hex()]; !ok {
+			spDetails, err := serviceProviderFactory.GetServiceProviderDetails(opts, node.Owner)
+			if err != nil {
+				return fmt.Errorf("failed get service provider details for address %s: %v", node.Owner.Hex(), err)
+			}
+			allServiceProviders[node.Owner.Hex()] = &db.EthServiceProvider{
+				Address:           node.Owner.Hex(),
+				DeployerStake:     spDetails.DeployerStake.Int64(),
+				DeployerCut:       spDetails.DeployerCut.Int64(),
+				ValidBounds:       spDetails.ValidBounds,
+				NumberOfEndpoints: int32(spDetails.NumberOfEndpoints.Int64()),
+				MinAccountStake:   spDetails.MinAccountStake.Int64(),
+				MaxAccountStake:   spDetails.MaxAccountStake.Int64(),
+			}
 		}
 	}
 
+	for _, sp := range allServiceProviders {
+		if err := txq.InsertServiceProvider(
+			ctx,
+			db.InsertServiceProviderParams{
+				Address:           sp.Address,
+				DeployerStake:     sp.DeployerStake,
+				DeployerCut:       sp.DeployerCut,
+				ValidBounds:       sp.ValidBounds,
+				NumberOfEndpoints: sp.NumberOfEndpoints,
+				MinAccountStake:   sp.MinAccountStake,
+				MaxAccountStake:   sp.MaxAccountStake,
+			},
+		); err != nil {
+			return fmt.Errorf("could not insert service provider into eth indexer db: %v", err)
+		}
+
+		totalStakedForSp, err := staking.TotalStakedFor(opts, ethcommon.HexToAddress(sp.Address))
```

Comment:
confirming that this includes delegated amount too? (I think it does, just want to make sure)